networkx/algorithms/centrality/tests/test_percolation_centrality.py

```diff
@@ -48,6 +72,23 @@ def test_percolation_example1b(self):
         for n, k in p_answer.items():
             assert p[n] == pytest.approx(k, abs=1e-3)
 
+    def test_percolation_example1c(self):
+        """percolation centrality: example 1c weighted edges"""
+        G = example1c_G()
+        p = nx.percolation_centrality(G)
+        p_answer = {
+            1: 0.0,
+            2: 0.0,
+            3: 0.0,
+            4: 0.825,
+            5: 0.5350877192982455,
+            6: 0.4,
+            7: 0.0,
+            8: 0.0,
+        }
```

Comment:
It is as a result of running percolation_centrality

Reply:
That's what I suspected :) - so the problem with doing that is that the test result is self-referential. In other words, what this *really* tests is that `percolation_centrality` gives the same answer as it does currently; whereas what we want is to test that `percolation_centrality` gives the *correct* answer.

Finding/designing such tests is definitely more challenging. The best source (IMO) is to find examples in the peer-reviewed literature, thus you can be reasonably confident that multiple people were involved in designing/reviewing the example. Failing that, the next best thing would be to work out the result yourself to verify that what the correct answer *should* be for the test case.

Reply:
Okay, will work on that now. I was a bit sick hence, the late replies.

Reply:
@rossbar the issue I have now is the papers I've found does not include calculation using weighted examples of percolation algorithm

Reply:
@rossbar please reply to my last message

Reply:
That's not surprising! Then you're on to option number 2:

> Failing that, the next best thing would be to work out the result yourself to verify that what the correct answer should be for the test case.

In principle the best thing to do is develop a simple test case that really probes the percolation centrality calculation, compute the expected results (*without* relying on the existing percolcation_centrality function), then implement the test case to verify you get the expected result.

Reply:
Alright. Alright. Thanks for your attention.

Reply:
@rossbar so I'm stuck here. I was going through the percolation centrality algorithm and trying to solve it step by step. At the _accumulate_percolation stage... the S parameter is not iterable. I don't think it's meant to be iterable in fact yet, parameter delta depends on it. 'delta = dict.fromkeys(S, 0)'. I don't know if you understand my complaints.

Reply:
S is an integer and as such, it also cannot be popped. my example works well when I calculate the percolation centrality directly but solving for it step by step.. is quite confusing

Reply:
From what I gathered, S is meant to be iterable. I don't know if it is my example or if there is a mistake cos S is meant to be S, P, sigma, _ = dijkstra(G, s, weight) where dijkstra is single_source_dijkstra_path right?

Reply:
Be careful -- the version of dijkstra being used is not `nx.single_source_dijkstra_path` it is a private helper function that appears in the module for betweenness centrality (imported at the top of `percolation.py`). So `S` is a list of the nodes discovered in a basic djikstra traversal of the graph.  So, it is a list.

Also, when the process is to "work out the result yourself", that doesn't mean that you should follow the code already written and do all the steps by hand. That just "runs the code" really slowly and in an error prone way.  The idea is to work out what the percolation centrality should be for a simple example starting from its definition and **not following the code that we're trying to test**. Said another way, we want to find a different method of computing the answer so that we can check whether our code's method agrees with that.   

fun! fun! :)

Reply:
Wow. This might take longer than I planned.